App auth method policy - trusted CA issuer docs (#21187)
microsoftgraph/microsoft-graph-docs: api-reference/beta/resources/keycredentialconfiguration.md

---
title: "keyCredentialConfiguration resource type"

> description: "Represents a key credential configuration object that contains properties to configure restrictions such as restricting the lifetime of key secrets."

Reviewer: I believe this object is for certs while passwordCredentialConfiguration is for secrets, no? Please confirm.
```suggestion
description: "Represents a key credential configuration object that contains properties to configure restrictions for application certificates."
```

[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]

> Represents a key credential configuration object that contains properties to configure restrictions such as restricting the lifetime of key secrets.

Reviewer: ```suggestion
Represents a key credential configuration object that contains properties to configure restrictions for application certificates.
```

## Properties

| Property                            | Type                                                                               | Description                                                                                                                                                                                                                                                                                   |
| :---------------------------------- | :--------------------------------------------------------------------------------- | :-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

> |certificateBasedApplicationConfigurationIds|String collection|Collection of GUIDs that point to the certificate chain that contains the collection of allowed root and intermediate certificate authorities.|

Reviewer: ```suggestion
|certificateBasedApplicationConfigurationIds|String collection|Collection of GUIDs that point to the [certificateBasedApplicationConfiguration](../resources/certificatebasedapplicationconfiguration.md) that contains the collection of allowed root and intermediate certificate authorities.|
```

| maxLifeTime                         | Duration                                                                           |Value that can be used as the maximum duration in days, hours, minutes, or seconds from the date of key creation, for which the key is valid.  Defined in ISO 8601 format for Durations. For example, `P4DT12H30M5S` represents a duration of four days, twelve hours, thirty minutes, and five seconds. This property is required when **restrictionType** is set to `keyLifetime`. |
| restrictForAppsCreatedAfterDateTime | DateTimeOffset                                                                     | Timestamp when the policy is enforced for all apps created on or after the specified date. For existing applications, the enforcement date would be back dated. To apply to all applications regardless of their creation date, this property would be `null`. Nullable. |

> | restrictionType                     | appKeyCredentialRestrictionType | The type of restriction being applied. Possible values are `asymmetricKeyLifetime`, `unknownFutureValue`. Each value of restrictionType can be used only once per policy.                                                                                                                        |

Reviewer: You added trustedCertificateAuthority enum member.
```suggestion
| restrictionType                     | appKeyCredentialRestrictionType | The type of restriction being applied. Possible values are `asymmetricKeyLifetime`, `trustedCertificateAuthority`, `unknownFutureValue`. Each value of restrictionType can be used only once per policy.                                                                                                                        |
```

```json
{
  "@odata.type": "#microsoft.graph.keyCredentialConfiguration",
  "certificateBasedApplicationConfigurationIds": ["String"],
  "maxLifetime": "String (duration)",
  "restrictForAppsCreatedAfterDateTime": "String (timestamp)",

>   "restrictionType": {"@odata.type": "microsoft.graph.appKeyCredentialRestrictionType"},

Reviewer: Little correction: enums are defined as "String" in the JSON representation.
```suggestion
  "restrictionType": "String",
```
